src/main/java/com/openlattice/datastore/search/controllers/SearchController.java

```diff
@@ -141,8 +143,53 @@ public SearchResult getEntitySets(
             produces = { MediaType.APPLICATION_JSON_VALUE } )
     @Override
     public DataSearchResult searchEntitySetData( @RequestBody SearchConstraints searchConstraints ) {
-
-        DataSearchResult results = searchService.executeSearch( searchConstraints, Principals.getCurrentPrincipals() );
+        // check read on entity sets
+        final var authorizedEntitySetIds = Arrays.stream( searchConstraints.getEntitySetIds() )
```

Comment:
Is all this authorization logic also present in SearchService? `SearchService. executeSearch` is also called from `PersistentSearchMessenger` for a particular user's principals, so authorization piece was originally moved into SearchService so that we didn't have to duplicate all this authorization logic in multiple places.

Reply:
I moved it out of SearchService.executeSearch to the SearchController and PersistentSearchMessengerTask, because far as I know, we prefer to do all the authentication in the controller classes and the logic for linking entity sets had to change anyway. 
Now SearchService.executeSearch takes only SearchConstraints searchConstraints and Map<UUID, Map<UUID, PropertyType>> authorizedPropertyTypesByEntitySet as inputs

Reply:
In that case I think it would be better to either move the logic back into SearchService or (preferable) add something like getAuthorizedPropertyTypesByEntitySetsForPrincipals that performs this logic in HazelcastAuthorizationService to avoid having to maintain this logic in multiple places.

Reply:
Okay, I will move it back then

Reply:
I think moving it into HazelcastAuthorizationService would be better actually -- this seems like something that might be useful in other places than this function

Reply:
Okay, I will create a function in EdmAuthorizationHelper

Reply:
Oh yeah that would be better than HazelcastAuthorizationService. Cool thanks!

Reply:
One question though: In SearchController.searchEntitySetData shouldn't we only allow the search if the calling user has read access on all of them? Or if we allow partial searches, shouldn't we change the code in SearchService.executeSearch to only do the search on authorized entitysets (right now it's using the entity set ids of the SearchConstraints)

Reply:
Yeah, I think it makes sense require read on all the entity sets but not require read on all property types